### ISSUE TITLE
[Issue] User Management: Delete Passage identity on user deactivation

### ISSUE THREAD

**Issue by TylerHendrickson** [usdigitalresponse/cpf-reporter#107]
**DRAFT: THIS TICKET IS NOT YET READY FOR WORK**

**Subtask of user story:** User Management https://github.com/usdigitalresponse/cpf-reporter/issues/81

**DOD:** When an update to an existing user sets `isActive=false`, the user's Passage identity is deleted.

**Blocked by issues:**
- https://github.com/usdigitalresponse/cpf-reporter/issues/103

**Blocks these issues:** (none)

**Implementation Details:**
- Update the `api/src/lib/auth.ts` module interface to provide an exported function that facilitates [deleting a user identity from Passage](https://docs.passage.id/backend/overview/node#user-management).
- Use the `lib/auth` module to create a new Passage user identity so that the following scenarios are satisfied:
   - In the `user` service's `updateUser` mutation resolver: when the request's `UpdateUserInput` input data will set `isActive=true` -> `isActive=false`, the user's `passageId` value is used to delete the user's Passage identity, and then set to `null` in the database.

**Comment by as1729** [comment]
Lower priority compared to registering a passage identity.

**Comment by ClaireValdivia** [comment]
@Vikariusu fyi that I created issue #381 for the front end changes for deactivating & reactivating a user

**Comment by ClaireValdivia** [comment]
Deactivation is working as expected, thanks so much for your work on this @Vikariusu !
